### ISSUE TITLE
Feature Request - Parameter Loading - MarginalCopula

### ISSUE THREAD

**Issue by kaleb-keny** [DanielBok/copulae#30]
I am not sure this is possible, but I am trying to do is save the parameters of the fit and load them in the future (as to avoid having to refit the data). So the parameters of a MarginalCopula fit are obtained using the following:
```
params = cop.params
Out[37]: 
{'copula': 6.935870895461638,
 'marginals': [{'type': 'gamma',
   'parameters': {'a': 1.2455705214968726,
    'loc': -0.00018850468681980907,
    'scale': 10.725816166928752}},
  {'type': 'genpareto',
   'parameters': {'c': 1.9383054484824958,
    'loc': -1.982576615044322e-09,
    'scale': 1.273334393803809}}]}
```
Then loading them back into a copula object is kind of tricky because the parameters for a Marginal Copula is a dict, as opposed to a numpy array. Maybe there is already a hard way to this.

**Comment by DanielBok** [comment]
One easy way is to use pickle

```python
from copulae.mixtures.gmc.parameter import GMCParam
import pickle

param = GMCParam(3, 2,
                 prob=[0.48923563, 0.05350188, 0.45726249],
                 means=[
                     [-7.866836, -11.073276],
                     [-3.336661, 24.139628],
                     [-23.879568, -5.213695]
                 ],
                 covs=[
                     [[15.9705696, -0.27545926],
                      [-0.2754593, 0.01738696]],
                     [[3.973527, -2.588777],
                      [-2.588777, 6.676961]],
                     [[5.3549889, -0.3326589],
                      [-0.3326589, 4.8624179]]
                 ])

# save the object, remember to use 'wb'
with open('my-file.p', 'wb') as f:
    pickle.dump(param, f)

# subsequently, you can load it back as such, remember to use 'rb'
with open('my-file.p') as f:
    my_params = pickle.load(f)
```

**Comment by DanielBok** [comment]
Good point, I'll think of how we can make it look as such for the GMC

```python
# assuming params is the dictionary, will do some checks to ensure it's properly defined here
cop.params = params
```

I believe one alternative to saving the copula parameters is to save the copula object as a pickle. This removes the issue with the user needing to know which copula the parameter is for.

```python
import pickle

with open('my-cop.p', 'wb') as f:
    pickle.dump(my_cop, f)
```
Later on, we wouldn't need to know whether to import Gaussian, Student, or any other copula could just load the copula as-is. The parameters can still be accessed via `copula.params` property

**Comment by kaleb-keny** [comment]
@DanielBok although that is the easy way out 😄 and I was thinking of pickling the cop object lol... But I thought that that wouldn't be like the most elegant of solutions... I mean, that saving & loading the parameters of the copula, which is kind of a dict now,  will allow transferring of parameters and allow you to see how the parameters are evolving across different fits... But maybe for this to work, you need maybe to add to the dict in cop.params the type of copula being used so on load it can push in those parameters appropriately... 
I mean the solution you proposed for the gaussian/student works perfectly as you just need that array (representing correlation matrix)... Something that follows along that kind of work would be a dict with parameters inside...
But I'll probably leave the design part to you, as you might know more about the complications of making a generalized copula class that takes in any parameters that is pre-saved and generates the copula accordingly....

**Comment by kaleb-keny** [comment]
Yeah I it ran without issue.. Although 😅  I will need to do some experimentation, as I just can't seem to get the same level of linearity as the one obtained empirically...
thank you though, appreciate that you build this package...
